### PR TITLE
Pulled duplicate code into PreferencesBase

### PR DESCRIPTION
**Pull request by nspruit**
Turned the interface `PreferencesBase` into an abstract class and pulled some duplicate code in the two `Preferences` classes into it.

Refs #219

### CODE REVIEW

File: WatchDogCore/WatchDog/src/nl/tudelft/watchdog/core/ui/preferences/PreferencesBase.java [hunk 1]
```diff
@@ -1,87 +1,131 @@
...
+	
+	/** The map of registered workspaces. */
+	protected List<ProjectPreferenceSetting> projectSettings = new ArrayList<ProjectPreferenceSetting>();
 	
 	/**
 	 * Returns whether logging is enabled (<code>true</code>) or not (
 	 * <code>false</code>).
 	 */
-    boolean isLoggingEnabled();
+    public abstract boolean isLoggingEnabled();
```
Comment (Inventitech) [inline WatchDogCore/WatchDog/src/nl/tudelft/watchdog/core/ui/preferences/PreferencesBase.java]: Thanks for adding the modifiers.

File: WatchDogCore/WatchDog/src/nl/tudelft/watchdog/core/ui/preferences/PreferencesBase.java [hunk 2]
```diff
@@ -1,87 +1,131 @@
...
 	/**
 	 * @return <code>true</code> if this workspace has already been registered
 	 *         with WatchDog, <code>false</code> otherwise. Note: This does not
 	 *         say whether WatchDog should be activated.}.
 	 */
-    boolean isProjectRegistered(String project);
+    public boolean isProjectRegistered(String project) {
+    	ProjectPreferenceSetting projectSetting = getProjectSetting(
+				project);
+		return (projectSetting != null
```
Comment (Inventitech) [inline WatchDogCore/WatchDog/src/nl/tudelft/watchdog/core/ui/preferences/PreferencesBase.java]: Legacy code, but parens do not make sense this way.

File: WatchDogCore/WatchDog/src/nl/tudelft/watchdog/core/ui/preferences/PreferencesBase.java [hunk 3]
```diff
@@ -1,87 +1,131 @@
...
+		for (ProjectPreferenceSetting setting : projectSettings) {
+			if (setting.project.equals(project)) {
+				return setting;
+			}
+		}
+		return null;
+	}
 
 	/**
 	 * Registers the given workspace with WatchDog. If use is <code>true</code>,
```
Comment (Inventitech) [inline WatchDogCore/WatchDog/src/nl/tudelft/watchdog/core/ui/preferences/PreferencesBase.java]: Replace workspace with project? Could you grep the whole code for occurrences of workspace? That should only appear in Eclipse plugin. Ideally, a short explanation on this difference should be in ...DogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/eclipse/ui/preferences/Preferences, for example in the JDoc.

**Comment by Inventitech** [comment]
Excellent! I really like your code. Just a few small comments.

**Comment by nspruit** [comment]
Comments have now been fixed in my opinion.

**Comment by Inventitech** [comment]
Nice, thanks for this PR!

**Pull request merged**